packages/inngest/src/components/InngestCommHandler.ts

```diff
@@ -889,22 +905,39 @@ export class InngestCommHandler<
           has_signing_key: Boolean(this.signingKey),
           function_count: registerBody.functions.length,
           mode: this._mode.type,
+          schema_version: "2024-05-24",
         };
 
-        // Only allow secure introspection in Cloud mode, since Dev mode skips
+        // Only allow authenticated introspection in Cloud mode, since Dev mode skips
         // signature validation
         if (this._mode.type === "cloud") {
           try {
             this.validateSignature(signature ?? undefined, "");
 
             introspection = {
               ...introspection,
-              signing_key_fallback_hash: this.hashedSigningKeyFallback,
-              signing_key_hash: this.hashedSigningKey,
-            };
+              authentication_succeeded: true,
+              api_origin: this.client["apiBaseUrl"] ?? defaultInngestApiBaseUrl,
+              app_id: this.client.id,
+              env: this.client["headers"][headerKeys.Environment] ?? null,
```

Comment:
(More fun) Some environment variables are only available to the comm handler and not to the client. For this, I'd use the `actions.headers()` function which will fetch it from the serve handler:
```suggestion
              env:
                (await actions.headers(
                  "fetching environment for introspection request",
                  headerKeys.Environment
                )) || null,
```